torch/csrc/utils/tensor_new.cpp

```diff
@@ -284,15 +307,27 @@ Tensor internal_new_from_data(
     maybe_initialize_cuda(device);
     return tensor.to(device, inferred_scalar_type, /*non_blocking=*/false, /*copy=*/copy_numpy);
   }
+
+  // We first check whether the numpy array pointers were extracted. If so, we continue with raw memory
+  // copying part : torch::utils::tensor_from_ndarray_batch.
+  // In case data contains anything other than all numpy arrays, this approach will not be taken and
+  // instead recusrive_store as shown below will be used for copying.
+  std::vector<PyArrayObject*> array_ptr_storage;
+  if (torch::utils::extract_ndarrays(data, sizes, array_ptr_storage)) {
+    at::AutoNonVariableTypeMode guard;
+    at::tracer::impl::NoTracerDispatchMode tracer_guard;
+    tensor = torch::utils::tensor_from_ndarray_batch(array_ptr_storage, sizes, inferred_scalar_type, pin_memory);
+    if (as_contiguous) {
+      tensor = tensor.contiguous();
```

Comment:
Would matching the behavior today require calling contiguous()?

Reply:
As per https://github.com/pytorch/pytorch/pull/51731#discussion_r593522402 , I have removed this part.